packages/firebase_messaging/firebase_messaging/android/src/main/java/io/flutter/plugins/firebase/messaging/FlutterFirebaseMessagingStore.java

```diff
@@ -20,7 +20,7 @@
 public class FlutterFirebaseMessagingStore {
   private static final String PREFERENCES_FILE = "io.flutter.plugins.firebase.messaging";
   private static final String KEY_NOTIFICATION_IDS = "notification_ids";
-  private static final int MAX_SIZE_NOTIFICATIONS = 20;
+  private static final int MAX_SIZE_NOTIFICATIONS = 100;
```

Comment:
Any reason we have a notification limit on Android and not on iOS?

Reply:
I think it is legacy from before we worked on this project.

Reply:
Is iOS 100 as well?

Reply:
messaging works differently on iOS. We don't have to store any messages, we call `onBackgroundMessages` method channel immediately. android is different, we have to spool up a Flutter engine before we make calls to the Dart side hence the message store on android.